packages/backend/src/plugins/scaffolder.ts

```diff
@@ -17,13 +17,17 @@
 import {
   CookieCutter,
   createRouter,
-  DiskStorage,
+  FilePreparer,
+  Preparers,
 } from '@backstage/plugin-scaffolder-backend';
 import type { PluginEnvironment } from '../types';
 
 export default async function createPlugin({ logger }: PluginEnvironment) {
-  const storage = new DiskStorage({ logger });
   const templater = new CookieCutter();
+  const filePreparer = new FilePreparer();
+  const preparers = new Preparers();
 
-  return await createRouter({ storage, templater, logger });
+  preparers.register('file', filePreparer);
```

Comment:
so something like:

```ts
const preparers = preparerers
    .register('github', github())
    .register('file', file())

preparers.get(entity)
```

Reply:
I like the long term idea of things being able to register themselves rather them being instantiated at construction though. Thought that it made for a better plugin API, but happy to be convinced otherwise.

Reply:
Yep, no strong opinions tbh